test/TesterInternal/TesterInternal.csproj

```diff
@@ -27,6 +27,7 @@
     <ProjectReference Include="..\..\src\Azure\Orleans.Persistence.AzureStorage\Orleans.Persistence.AzureStorage.csproj" />
     <ProjectReference Include="..\..\src\Orleans.CodeGeneration\Orleans.CodeGeneration.csproj" />
     <ProjectReference Include="..\..\src\Orleans.Core.Legacy\Orleans.Core.Legacy.csproj" />
+    <ProjectReference Include="..\..\src\Orleans.TestingHost.Legacy\Orleans.TestingHost.Legacy.csproj" />
```

Comment:
why are we fall back to use TestingHost.Legacy? we shouldn't be adding tests using this package. The end goal is to move away from it and also testing the new `TestingHost` patterns

Reply:
`ClientInitTests` uses it, that's unchanged but I moved the class from DefaultCluster.Tests to here, since it doesn't use the default cluster fixture.

This PR moved almost every test in DefaultCluster.Tests away from legacy.